src/components/product.js

```diff
@@ -163,7 +173,8 @@ export default class Product extends Component {
    * @return {Object} viewData object.
    */
   get viewData() {
-    return merge(this.model, this.options.viewData, {
+    const model = Object.assign({}, this.model);
+    return merge(model, this.options.viewData, {
```

Comment:
my question is why you don't just do `Object.assign({}, this.model, this.options.viewData, { ... })`